### ISSUE TITLE
FR: Be able to use any github repo without kurtosis locators

### ISSUE THREAD

**Issue by barnabasbusa** [kurtosis-tech/kurtosis#2216]
### Background & motivation

It would be great if I could reference any github repo without needing to have a kurtosis.yml locator. 

### Desired behaviour

Be able to use any github repo without kurtosis locator. 

### How important is this to you?

Painful; the lack of this feature makes using Kurtosis frictionful.

### What area of the product does this pertain to?

CLI: the Command Line Interface

**Comment by mieubrisse** [comment]
Thanks for filing this @barnabasbusa ! I'd love to get more info on the usecase you're thinking about for this

**Comment by barnabasbusa** [comment]
it would be in combination of this issue: https://github.com/kurtosis-tech/kurtosis/issues/2215

Where I could reference any arbitrary dockerfile in any 
random github repo, or possibly any url.

**Comment by mieubrisse** [comment]
This makes sense to me. So just repeating my understanding of this ticket + #2215 , would the following solve your problem?
1. Kurtosis allows you to use Github locators even if there's not a Kurtosis `kurtosis.yml` at it (e.g. `github.com/ethereum/go-ethereum`)
2. Kurtosis allows you to use arbitrary paths on your filesystem (e.g. `/home/barnabas/code/my-custom-microservice`)

IMPLEMENTATION NOTES FOR KURTOSIANS:
- The above supposes "implicit packages", i.e. we don't need an explicit `kurtosis.yml`
- If we do this, we should have a lockfile to ensure that Kurtosians can't be victim of supply chain attacks. This has been a long time in coming anyways.

**Comment by barnabasbusa** [comment]
Yes, and ideally also allow random urls and use the content of the raw url to build a dockerimage. 

e.g pastebin url
